backend/coins/eth/account.go

```diff
@@ -285,19 +287,41 @@ type TxProposal struct {
 
 func (account *Account) newTx(
 	recipientAddress string,
-	amount coin.SendAmount) (*TxProposal, error) {
+	amount coin.SendAmount,
+	data []byte,
+) (*TxProposal, error) {
 	if !common.IsHexAddress(recipientAddress) {
 		return nil, errp.WithStack(coin.ErrInvalidAddress)
 	}
-	const gasLimit = 21000 // simple transaction gas cost
 
 	suggestedGasPrice, err := account.coin.client.SuggestGasPrice(context.TODO())
 	if err != nil {
 		return nil, err
 	}
-	fee := new(big.Int).Mul(big.NewInt(gasLimit), suggestedGasPrice)
 
-	var value *big.Int
+	parsedAmount, err := amount.Amount(big.NewInt(params.Ether))
```

Comment:
you do the same below in the else branch, which can be removed (also `value := parsedAmount.BigInt()`)